rpc_jobs/build_gating_venv.yml

```diff
@@ -20,102 +20,99 @@
         lock('build_gating_venv'){
           node('pubcloud_multiuse'){
             try{
-              deleteDir()
-              common.clone_with_pr_refs('rpc-gating',
-                                        'git@github.com:rcbops/rpc-gating',
-                                        env.RPC_GATING_BRANCH)
-              dir('rpc-gating'){
-                // run in docker to ensure all the required apt packages are available
-                container = docker.build env.BUILD_TAG.toLowerCase()
-              }
-              // Venv is build within docker as it requires extra packages
-              container.inside{
-                sh """#!/bin/bash -xeu
+              withCredentials(artifact_build.get_rpc_repo_creds()) {
```

Comment:
Is this needed by anything other than the shell script? If not, I'd say only wrap it around what's required.

Reply:
There's a note that says:

```
// venv is pushed to rpc repo from outside docker container, as binding
// credentials files does not work within docker (used for ssh key)
```

... which explains why this has to wrap the container.